tasks/main.yml

```diff
@@ -5,23 +5,160 @@
 - name: Set platform/version specific variables
   include_tasks: tasks/set_vars.yml
 
-# Examples of some tasks:
+- name: Gather the package facts
+  ansible.builtin.package_facts:
+    manager: auto
+
+- name: Check if requested version is supported in the system (RHEL8)
+  fail:
+    msg: RHEL 8 supports only Postgresql 10, 12 and 13
+  when:
+    - ansible_facts["os_family"] == "RedHat"
+    - ansible_facts["distribution_major_version"] == "8"
+    - postgresql_version not in __postgresql_versions_el8
+
+- name: Check if requested version is supported in system (RHEL9)
+  fail:
+    msg: RHEL 9 supports only Postgresql 13
+  when:
+    - ansible_facts["os_family"] == "RedHat"
+    - ansible_facts["distribution_major_version"] == "9"
+    - postgresql_version not in __postgresql_versions_el9
+
+- name: Check requested and installed version of Postgresql
```

Comment:
Should the role be able to upgrade from an earlier postgres?

Reply:
It isn't supported case